### PR TITLE
Refactor crosscheck

### PR DESCRIPTION
**Pull request by gitMarky**
Please also read the commit messages, my code modifications may make more sense if you do :)

### CODE REVIEW

File: src/object/C4GameObjects.cpp [hunk 1]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
 							// Only hit if obj2's speed and relative speeds are larger than HitSpeed2
-							if ((obj2->OCF & OCF_HitSpeed2) && speed > HitSpeed2 &&
-							   !obj1->Call(PSF_QueryCatchBlow, &C4AulParSet(obj2)))
+							if ((ball->OCF & OCF_HitSpeed2) && speed > HitSpeed2 &&
+							   !goal->Call(PSF_QueryCatchBlow, &C4AulParSet(ball)))
 							{
-								int32_t iHitEnergy = fixtoi(speed * obj2->Mass / 5);
+								int32_t hit_energy = fixtoi(speed * ball->Mass / 5);
 								// Hit energy reduced to 1/3rd, but do not drop to zero because of this division
-								iHitEnergy = std::max<int32_t>(iHitEnergy/3, !!iHitEnergy);
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: Wow, that was surprisingly awful.

File: src/object/C4GameObjects.cpp [hunk 2]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
-								obj1->DoEnergy(-iHitEnergy / 5, false, C4FxCall_EngObjHit, obj2->Controller);
-								int tmass = std::max<int32_t>(obj1->Mass, 50);
-								C4PropList* pActionDef = obj1->GetAction();
+								hit_energy = std::max<int32_t>(hit_energy/3, !!hit_energy);
+								goal->DoEnergy(-hit_energy / 5, false, C4FxCall_EngObjHit, ball->Controller);
+								int tmass = std::max<int32_t>(goal->Mass, 50);
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: I think `tmass` should be `goal_mass` with your naming scheme.

File: src/object/C4GameObjects.cpp [hunk 3]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
+							   !goal->Call(PSF_QueryCatchBlow, &C4AulParSet(ball)))
 							{
-								int32_t iHitEnergy = fixtoi(speed * obj2->Mass / 5);
+								int32_t hit_energy = fixtoi(speed * ball->Mass / 5);
 								// Hit energy reduced to 1/3rd, but do not drop to zero because of this division
-								iHitEnergy = std::max<int32_t>(iHitEnergy/3, !!iHitEnergy);
-								obj1->DoEnergy(-iHitEnergy / 5, false, C4FxCall_EngObjHit, obj2->Controller);
-								int tmass = std::max<int32_t>(obj1->Mass, 50);
-								C4PropList* pActionDef = obj1->GetAction();
+								hit_energy = std::max<int32_t>(hit_energy/3, !!hit_energy);
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: Can you change this to a slightly more readable
```cpp
if (hit_energy != 0) {
	hit_energy = std::max(hit_energy / 3, 1);
}
```

File: src/object/C4GameObjects.cpp [hunk 4]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
-				{
-					// Search match
-					if (cocf & ocf) { ocf=cocf; return cObj; }
-					// EXCLUSIVE block
-					else return nullptr;
 				}
 			}
-	return nullptr;
+			// Goto-marker for more obvious loop-control
+			check_next_object: ;
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: Which one? Next `goal` or next `ball`?

File: src/object/C4GameObjects.cpp [hunk 5]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
-C4Object *C4GameObjects::SafeObjectPointer(int32_t iNumber)
+C4Object *C4GameObjects::SafeObjectPointer(int32_t object_number)
 {
-	C4Object *pObj = ObjectPointer(iNumber);
-	if (pObj) if (!pObj->Status) return nullptr;
-	return pObj;
+	C4Object *game_object = ObjectPointer(object_number);
+	if (game_object)
+	{
+		if (!game_object->Status)
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: You can `&&` the conditions to reduce the indentation depth

File: src/object/C4GameObjects.cpp [hunk 6]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
 	}
 
-	// Denumerate pointers
-	// on section load, inactive object numbers will be adjusted afterwards
-	// so fake inactive object list empty meanwhile
-	// note this has to be done to prevent even if object numbers did not collide
-	// to prevent an assertion fail when denumerating non-enumerated inactive objects
+	// Denumerate pointers:
+	// On section load, inactive object numbers will be adjusted afterwards,
+	// so fake inactive object list empty. Meanwhile, note this has to be done
```
Comment (isilkor) [inline src/object/C4GameObjects.cpp]: The meanwhile should modify "fake inactive object list", not the note.

File: src/object/C4GameObjects.cpp (hunk 1 above)
Comment (gitMarky) [reply src/object/C4GameObjects.cpp]: What do you mean?

File: src/object/C4GameObjects.cpp [hunk 7]
```diff
@@ -46,355 +46,401 @@ void C4GameObjects::Default()
...
-	ForeObjects.Remove(pObj);
-	// manipulate main list
-	return C4ObjectList::Remove(pObj);
-}
-
-C4ObjectList &C4GameObjects::ObjectsAt(int ix, int iy)
+bool C4GameObjects::Remove(C4Object *game_object)
 {
-	return Sectors.SectorAt(ix, iy)->ObjectShapes;
+	// If it's an inactive object, simply remove from the inactiv elist
```
[the remark this answers is not in the document]
Comment (gitMarky) [reply src/object/C4GameObjects.cpp]: Done

File: src/object/C4GameObjects.cpp (hunk 2 above)
Comment (gitMarky) [reply src/object/C4GameObjects.cpp]: Done.

File: src/object/C4GameObjects.cpp (hunk 1 above)
Comment (gitMarky) [reply src/object/C4GameObjects.cpp]: Got it

**Pull request merged**